romancal/dq_init/tests/test_dq_init.py

```diff
@@ -0,0 +1,262 @@
+    ref_data = testutil.mk_mask(csize[1:])
+    ref_data['meta']['instrument']['name'] = instrument
+
+    # run the correction step
+    outfile = do_dqinit(dm_ramp, ref_data)
+
+    # check that GROUPDQ was created and initialized to zero
+    groupdq = outfile.groupdq
+
+    np.testing.assert_array_equal(np.full((ngroups, ysize, xsize), 0, dtype=int),
+                                  groupdq, err_msg='groupdq not initialized to zero')
+
+
+def test_err():
+    """Check that a 4-D ERR array is initialized and all values are zero."""
+
+    # size of integration
+    instrument = 'WFI'
+    ngroups = 5
+    xsize = 1032
+    ysize = 1024
+    csize = (ngroups, ysize, xsize)
+
+    # create raw input data for step
+    dm_ramp = testutil.mk_ramp((ngroups, ysize, xsize))
+    dm_ramp.meta.instrument.name = instrument
+
+    # create a MaskModel elements for the dq input mask
+    ref_data = testutil.mk_mask(csize[1:])
+    ref_data['meta']['instrument']['name'] = instrument
+
+    # Filter out validation warnings from ref_data
+    warnings.filterwarnings("ignore", category=ValidationWarning)
+
+    # run correction step
+    outfile = do_dqinit(dm_ramp, ref_data)
+
+    # check that ERR array was created and initialized to zero
+    errarr = outfile.err
+
+    assert(errarr.ndim == 2)  # check that output err array is 2-D
+    assert(np.all(errarr == 0))  # check that values are 0
+
+
+def test_dq_add1_groupdq():
+    """
+    Test if the dq_init code set the groupdq flag on the first
+    group to 'do_not_use' by adding 1 to the flag, not overwriting to 1
+    Also test whether two flags on the same pixel are added together.
+    """
+
+    # size of integration
+    instrument = 'WFI'
+    ngroups = 5
+    xsize = 1032
+    ysize = 1024
+    csize = (ngroups, ysize, xsize)
+
+    # create raw input data for step
+    dm_ramp = testutil.mk_ramp((ngroups, ysize, xsize))
+    dm_ramp.meta.instrument.name = instrument
+
+    # create a MaskModel elements for the dq input mask
+    dq = np.zeros(csize[1:], dtype=np.uint32)
+
+    # write reference file with known bad pixel values
+    dq[505, 505] = 1   # Do_not_use
+    dq[400, 500] = 3  # do_not_use and saturated pixel
+
+    # write mask model
+    ref_data = testutil.mk_mask(csize[1:])
+    ref_data['meta']['instrument']['name'] = instrument
+
+    # Copy in maskmodel elemnts
+    ref_data['dq'] = dq
+
+    # set a flag in the pixel dq
+    dm_ramp.pixeldq[505, 505] = 4 # Jump detected pixel
+
+    # run correction step
+    outfile = do_dqinit(dm_ramp, ref_data)
+
+    # test if pixels in pixeldq were incremented in value by 1
+    # check that previous dq flag is added to mask value
+    assert(outfile.pixeldq[505, 505] == dqflags.pixel['JUMP_DET'] + dqflags.pixel['DO_NOT_USE'])
+    # check two flags propagate correctly
+    assert(outfile.pixeldq[400, 500] == dqflags.pixel['SATURATED'] + dqflags.pixel['DO_NOT_USE'])
+
+
+# Commented out until CRDS updated with mask information
```

Comment:
The commented oput test below should be a regression test. There's no access to CRDS from teh unit tests suit and this test is more appropriate as a regression test anyway.

Reply:
Okay - can I spin that off into its own ticket? (Seems no point in making a regression test in this ticket, as there are no masks in CRDS.)

Reply:
Yes, start a new PR.

Reply:
Made an issue for it (#265 )

Reply:
Removed.